test_e2e_arc/arc_jobs_test.go

```diff
@@ -0,0 +1,132 @@
+package e2e_dev_arc
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"net/http"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+type podCountsByType struct {
+	controllers int
+	listeners   int
+	runners     int
+}
+
+func getPodsByType(clientset *kubernetes.Clientset) podCountsByType {
+	arc_namespace := "arc-systems"
+	availableArcPods, err := clientset.CoreV1().Pods(arc_namespace).List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		panic(err.Error())
+	}
+	runners_namespace := "arc-runners"
+	availableRunnerPods, err := clientset.CoreV1().Pods(runners_namespace).List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		panic(err.Error())
+	}
+	podsByType := podCountsByType{}
+	for _, pod := range availableArcPods.Items {
+		if strings.Contains(pod.Name, "controller") {
+			podsByType.controllers += 1
+		}
+		if strings.Contains(pod.Name, "listener") {
+			podsByType.listeners += 1
+		}
+	}
+	for _, pod := range availableRunnerPods.Items {
+		if strings.Contains(pod.Name, "runner") {
+			podsByType.runners += 1
+		}
+	}
+	return podsByType
+}
+
+func pollForClusterState(clientset *kubernetes.Clientset, expectedPodsCount podCountsByType, maxTime int) bool {
+	sleepTime := 5
+	maxRetries := maxTime / sleepTime
+	success := false
+	for i := 0; i <= maxRetries; i++ {
+		time.Sleep(time.Second * time.Duration(sleepTime))
+		availablePodsCount := getPodsByType(clientset)
+		if availablePodsCount == expectedPodsCount {
+			success = true
+			break
+		} else {
+			fmt.Printf("%v", availablePodsCount)
+		}
+	}
+	return success
+}
+
+func TestARCJobs(t *testing.T) {
+	configFile := filepath.Join(
+		os.Getenv("HOME"), ".kube", "config",
+	)
+
+	config, err := clientcmd.BuildConfigFromFlags("", configFile)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	t.Run("Get available pods before job run", func(t *testing.T) {
+		expectedPodsCount := podCountsByType{1, 1, 0}
+		success := pollForClusterState(clientset, expectedPodsCount, 60)
+		if !success {
+			t.Fatal("Expected pods count did not match available pods count before job run.")
+		}
+	},
+	)
+	t.Run("Get available pods during job run", func(t *testing.T) {
+		c := http.Client{}
+		dateTime := os.Getenv("DATE_TIME")
+		// We are triggering manually a workflow that already exists in the repo.
+		// This workflow is expected to spin up a number of runner pods matching the runners value set in podCountsByType.
+		url := "https://api.github.com/repos/actions/actions-runner-controller/actions/workflows/48368242/dispatches"
```

Comment:
Why don't we use: https://github.com/cli/go-gh instead of hardcoding the workflow ID here?

Reply:
Each one of ID, name, path will be hardcoded and any of these can change. If the name of the workflow changes, the ID and path stay the same. If the path changes, the ID will change. I will add the path for this reason and for readability! Thanks!

Reply:
Updated 🎊